geonode/security/models.py

```diff
@@ -388,3 +388,46 @@ def set_workflow_perms(self, approved=False, published=False):
             if settings.OGC_SERVER['default'].get("GEOFENCE_SECURITY_ENABLED", False):
                 if self.polymorphic_ctype.name == 'layer':
                     sync_geofence_with_guardian(self.layer, VIEW_PERMISSIONS)
+
+    def get_user_perms(self, user):
+        """
+        Returns a list of permissions a user has on a given resource
+        """
+        resource = self.get_self_resource()
+        user_resource_perms = get_users_with_perms(resource).get(user, [])
+        try:
+            if hasattr(self, "layer"):
+                user_resource_perms.extend(
+                    get_users_with_perms(self.layer).get(user, [])
```

Comment:
why do we need this? what is this adding to the previously collected user_resource_perms?

Reply:
This adds permissions that are defined in the Layer [here](https://github.com/GeoNode/geonode/blob/master/geonode/layers/models.py#L159)